tests/chainer_tests/links_tests/connection_tests/test_peephole.py

```diff
@@ -0,0 +1,206 @@
+        self.check_forward(cuda.to_gpu(self.c),
+                           cuda.to_gpu(self.h),
+                           cuda.to_gpu(self.x))
+
+    def check_backward(self, c_data, h_data, x_data, y_grad):
+        x = chainer.Variable(x_data)
+        y = self._forward(self.link, x)
+        y.grad = y_grad
+        y.backward()
+
+        def f():
+            c, y = _peephole(self.link, c_data, h_data, x_data)
+            return y,
+        gx, = gradient_check.numerical_grad(f, (x.data,), (y.grad,))
+        gradient_check.assert_allclose(gx, x.grad, atol=1e-3)
+
+    def test_backward_cpu(self):
+        self.check_backward(self.c, self.h, self.x, self.gy)
+
+    @attr.gpu
+    def test_backward_gpu(self):
+        self.link.to_gpu()
+        self.check_backward(cuda.to_gpu(self.c),
+                            cuda.to_gpu(self.h),
+                            cuda.to_gpu(self.x),
+                            cuda.to_gpu(self.gy))
+
+
+class TestPeepholeState(unittest.TestCase):
+
+    def setUp(self):
+        in_size, out_size = 10, 8
+        self.link = links.Peephole(in_size, out_size)
+
+    def check_reset_state(self):
+        self.link.reset_state()
+        self.assertIsNone(self.link.c)
+        self.assertIsNone(self.link.h)
+
+    def test_reset_state_cpu(self):
+        self.check_reset_state()
+
+    @attr.gpu
+    def test_reset_state_gpu(self):
+        self.link.to_gpu()
+        self.check_reset_state()
+
+
+class TestPeepholeToCPUToGPU(unittest.TestCase):
+
+    def setUp(self):
+        in_size, out_size = 10, 8
+        self.link = links.Peephole(in_size, out_size)
+        self.c = chainer.Variable(
+            numpy.random.uniform(-1, 1, (1, out_size)).astype(numpy.float32))
+        self.h = chainer.Variable(
+            numpy.random.uniform(-1, 1, (1, out_size)).astype(numpy.float32))
+
+    def check_to_cpu(self, c, h):
+        self.link.c = c
+        self.link.h = h
+        self.link.to_cpu()
+        self.assertIs(self.link.xp, numpy)
+        self.assertIsInstance(self.link.c.data, self.link.xp.ndarray)
+        self.assertIsInstance(self.link.h.data, self.link.xp.ndarray)
+        self.link.to_cpu()
+        self.assertIs(self.link.xp, numpy)
+        self.assertIsInstance(self.link.c.data, self.link.xp.ndarray)
+        self.assertIsInstance(self.link.h.data, self.link.xp.ndarray)
+
+    def test_to_cpu_cpu(self):
+        self.check_to_cpu(self.c, self.h)
+
+    @attr.gpu
+    def test_to_cpu_gpu(self):
+        self.c.to_gpu()
+        self.h.to_gpu()
+        self.check_to_cpu(self.c, self.h)
+
+    def check_to_cpu_to_gpu(self, c, h):
+        self.link.c = c
+        self.link.h = h
+        self.link.to_gpu()
+        self.assertIsNot(self.link.xp, numpy)
+        self.assertIsInstance(self.link.c.data, self.link.xp.ndarray)
+        self.assertIsInstance(self.link.h.data, self.link.xp.ndarray)
+        self.link.to_gpu()
+        self.assertIsNot(self.link.xp, numpy)
```

Comment:
This is something I want to ask you. At first, I wrote `assertIs(self.link.xp, cupy)` and of course `import cupy` at the top . However, when I pushed the code, Travis CI rejected it outputting error message `Cupy is not correctly installed` . The detailed image is as below. What should I do?
![screen shot 2016-06-18 at 03 04 54](https://cloud.githubusercontent.com/assets/8463142/16160233/030c02ca-3502-11e6-9626-ff6dd10e4701.png)